### PR TITLE
Fix #66 - Quick start features

### LINKED ISSUE #66: Add a getting started section
We should add some getting started section with a curated list of getting started resources. 

Should be able to vote up and down resources.

### PR DESCRIPTION
**Pull request by brunnoattorre**
Issue #66: Adding the quick start guides and allowing users to create and upvote/downvote guides.

### CODE REVIEW

File: test/ApplicationSpec.scala [hunk 1]
```diff
@@ -27,5 +27,10 @@ class ApplicationSpec extends Specification {
       contentAsString(home) must contain("GitRank")
       contentType(home) must beSome.which(_ == "text/html")
     }
+
+    "get list of guides" in new WithApplication{
+      val guides = route(FakeRequest(GET, "/github/angular/angular/quickstart/guides")).get
```
Comment (callicles) [inline test/ApplicationSpec.scala]: Codacy Issue found: [Usage of get on optional type.](https://www.codacy.com/app/gitlinks/gitrank-web/file/1092926018/issues/source?bid=2432310&fileBranchId=2891663#l32)

**Comment by pocman** [comment]
It's working for me.

**Comment by brunnoattorre** [comment]
@callicles issue on codacy fixed :+1:

File: app/controllers/ApplicationController.scala [hunk 2]
```diff
@@ -95,18 +98,104 @@ class ApplicationController @Inject()(
...
+          repo,
+          form.data.getOrElse("title", ""),
+          form.data.getOrElse("description", ""),
+          form.data.getOrElse("url", "")
+        ))
+        case None => Future(NotFound(views.html.error("notFound", 404, "Not Found",
+          "We cannot find the repository feedback page, it is likely that you misspelled it, try something else !")))
+      },
+      data => {
+        println("is it working?")
```
Comment (callicles) [inline app/controllers/ApplicationController.scala]: ? What is that ?

File: app/controllers/ApplicationController.scala [hunk 3]
```diff
@@ -95,18 +98,104 @@ class ApplicationController @Inject()(
...
+  }
+
+  /**
+   * Service for getting the quickstart guides of a repo
+   *
+   * @param owner Owner of the repository on the repo system (GitHub)
+   * @param repositoryName repository name on the repo system (GitHub)
+   * @return the list of guides for the given repo
+   */
+  def getGuides(owner: String, repositoryName: String) = UserAwareAction.async { implicit request =>
```
Comment (callicles) [inline app/controllers/ApplicationController.scala]: This is a method that doesn't require user connection and that returns Json, I think it should be moved in the PublicAPIController

File: app/controllers/ApplicationController.scala [hunk 4]
```diff
@@ -95,18 +98,104 @@ class ApplicationController @Inject()(
...
+  /**
+   * Service for upvoting a guide
+   *
+   * @param owner Owner of the repository on the repo system (GitHub)
+   * @param repositoryName repository name on the repo system (GitHub)
+   * @param title title of the guide
+   * @param voteType if the vote is upvote or downvote
+   * @return the guide
+   */
+  def upvote(owner: String, repositoryName: String, title: String, voteType: String) = UserAwareAction.async { implicit request =>
```
Comment (callicles) [inline app/controllers/ApplicationController.scala]: We request that the user is connected to upVote, right ? is UserAware Good enough ?

Comment (brunnoattorre) [inline app/controllers/ApplicationController.scala]: Yes. If the user is not logged in he won't be able to upvote/downvote. Also, he can only vote once (we have to review if we allow a user to "undo" his vote).

**Comment by callicles** [comment]
Url field in the form is not validated as such

**Comment by brunnoattorre** [comment]
Validations added in both FE and BE.

**Pull request merged**